### PR TITLE
Issue #11 Use string cache for Id<T>.

### PR DESCRIPTION
**Pull request by azasypkin**

### CODE REVIEW

File: src/selector.rs [hunk 1]
```diff
@@ -415,9 +417,9 @@ impl Period {
     }
 }
 
-fn has_selected_tags(actual: &HashSet<String>, requested: &HashSet<String>) -> bool {
+fn has_selected_tags(actual: &HashSet<String>, requested: &HashSet<Atom>) -> bool {
```
Comment (azasypkin) [inline src/selector.rs]: Hey @Yoric what is your plan on using `Atoms` - do you think we should use atoms for taxonomy-owned data only (e.g. service and service's getters and setters tags are atoms, but service selectors aren't), or API consumers should provide atoms as well (service selector tags also should be atoms)?

Comment (Yoric) [inline src/selector.rs]: I believe that tags should be atoms, too.

File: src/selector.rs [hunk 2]
```diff
@@ -415,9 +417,9 @@ impl Period {
...
     for tag in &*actual {
-        if !requested.contains(tag) {
+        if !requested.contains(&Atom::from(tag as &str)) {
```
Comment (Yoric) [inline src/selector.rs]: ... and since tags are atoms, this becomes much simpler/faster.

**Comment by Yoric** [comment]
Good start :)

**Comment by azasypkin** [comment]
> Good start :)

Ha-ha, especially failed Travis for such a tiny PR :)

File: src/selector.rs (hunk 2 above)
Comment (azasypkin) [inline src/selector.rs]: Right, thanks, that's why I'm asking :)

**Comment by azasypkin** [comment]
Hmm it worked well with my Cargo.lock :) Looking.

**Comment by azasypkin** [comment]
It seems we can't implement `Serialize` for `Atom` as neither of them is in our crate :) I started to play with so called `TaxonomyAtom` - wrapper around `Atom` until I find a better solution :)

**Comment by Yoric** [comment]
Let's start by making only `Id<T>` use `Atom`. Then we'll migrate stuff that needs to be atomicized to `Id<T>`.

**Comment by Yoric** [comment]
Looks good to me, thanks.

And thanks for taking the opportunity to advance #14 :)

**Comment by Yoric** [comment]
Also, you are now officially in charge of discussing with @SimonSapin to find a way to not bloat our binary with useless strings :)

**Comment by azasypkin** [comment]
> Looks good to me, thanks.
> And thanks for taking the opportunity to advance #14 :)

Np! You're so fast :) I was just about to ask if we want `Id<T>::new` to accept `Atom` instead of `String` as an argument :)

> Also, you are now officially in charge of discussing with @SimonSapin to find a way to not bloat our binary with useless strings :)

Sure :)

**Pull request merged**